nautobot/docs/plugins/development.md

```diff
@@ -609,6 +610,111 @@ secrets_providers = [ConstantValueSecretsProvider]
 
 After installing and enabling your plugin, you should now be able to navigate to `Secrets > Secrets` and create a new Secret, at which point `"constant-value"` should now be available as a new secrets provider to use.
 
+### Extending Filters
+
+Plugins can extend the current FilterSet's and FilterForm's that are natively provided.
+
+The basic requirements (additional requirements for each are described below) to extend a filter set or a filter form are:
+* The file must be named `filter_extensions.py`
+* The variable `filter_extensions` must be declared in that file, and contain a list of `PluginFilterExtension` subclasses
+* The `model` attribute of each `PluginFilterExtension` subclass must be set to a valid model name
+
+Nautobot already dynamically creates many filters for each field type which allows the extension of related fields. Specfically, there are additional `lookup_expr` that are created when there is neither a `lookup_expr` nor `method` parameter set (in this context, `method` is a filterset parameter, not a class method), with the specific expressions based on the type of `django-filter` field used such as `CHAR` vs `NUMERIC` as hinted in the variable definitions seen below. Additionally, Nautobot will add the negation of any fields, meaning not only will `icontains` be added but so will `not icontains` using the `ic` and `nic` keys respectively.
```

Comment:
This paragraph is really dense and technical - it might be worth revising, though I don't have a specific suggestion for doing so.